qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDatabaseQueries.groovy

```diff
@@ -342,31 +342,83 @@ class CustomerDatabaseQueries {
         return result
     }
 
+    /**
+     * Add an affiliation to the database
+     *
+     * @param affiliation which needs to be added to the database
+     */
     void addAffiliation(Affiliation affiliation) throws DatabaseQueryException {
-        log.debug("Adding affiliation " + affiliation.organisation + " " + affiliation.addressAddition + "to database")
-
-        /*
-        if (affiliationExists) {
+        if (affiliationExists(affiliation)) {
             throw new DatabaseQueryException("Affiliation is already in the database.")
         }
         Connection connection = databaseSession.getConnection()
         connection.setAutoCommit(false)
 
         connection.withCloseable {it ->
             try {
-                int customerId = createNewCustomer(it, customer)
-                storeAffiliation(it, customerId, customer.affiliations)
+                createNewAffiliation(it, affiliation)
                 connection.commit()
             } catch (Exception e) {
                 log.error(e.message)
                 log.error(e.stackTrace.join("\n"))
                 connection.rollback()
                 connection.close()
-                throw new DatabaseQueryException("Could not create customer.")
+                throw new DatabaseQueryException("Could not create affiliation.")
```

Comment:
Do you think it makes sense to also catch `DatabaseQueryException`s separately and forward a new exception with the message of the DatabaseQueryException? Otherwise, this will always throw an exception with the completely generic message. I do not know if this is desired or not.

Reply:
I did add a new exception but i'm also unsure if that's really necessary. If so this should also be added to the customer creation method